android/rustls-platform-verifier/src/main/java/org/rustls/platformverifier/CertificateVerifier.kt

```diff
@@ -330,7 +330,9 @@ internal object CertificateVerifier {
 
             revocationChecker.options = EnumSet.of(
                 PKIXRevocationChecker.Option.SOFT_FAIL,
-                PKIXRevocationChecker.Option.ONLY_END_ENTITY
+                PKIXRevocationChecker.Option.ONLY_END_ENTITY,
+                PKIXRevocationChecker.Option.PREFER_CRLS,
+                PKIXRevocationChecker.Option.NO_FALLBACK
```

Comment:
As some immediate feedback, nonwithstanding the newly posted information about Android's internal behavior, this part would need to be gated behind `if (!BuildConfig.TEST)` in order to support our tests that validate stapled OCSP data.

Despite OCSP [going on its way out](https://www.feistyduck.com/newsletter/issue_121_the_slow_death_of_ocsp) I don't think the tests are removable yet because its stable-ish behavior and I don't see a reason to completely drop them yet.